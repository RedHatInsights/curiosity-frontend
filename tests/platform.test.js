const { execSync } = require('child_process');
const packageJson = require('../package');

describe('Platform Configuration, Build', () => {
  it('should have a name reference that matches configuration', () => {
    const envFile = './.env';
    const { appname } = packageJson.insights;

    const output = execSync(`grep -w "REACT_APP_UI_NAME=${appname}" ${envFile}`);
    expect(output.toString().split('\n')).toMatchSnapshot('ui name');
  });

  it('should have path references that match configuration', () => {
    const envFile = './.env';
    const { appname } = packageJson.insights;

    const output = execSync(`grep "\\/${appname}\\/" ${envFile}`);
    expect(output.toString().split('\n')).toMatchSnapshot('path references');
  });

  it('should have a proxy configuration with path references that match configuration', () => {
    const proxyFile = './config/spandx.config.js';
    const { appname } = packageJson.insights;

    const output = execSync(`grep "\\/${appname}'" ${proxyFile}`);
    expect(
      output
        .toString()
        .split('\n')
        .map(value => value.trim())
    ).toMatchSnapshot('proxy references');
  });

  it('should use direct imports for platform components, with exceptions', () => {
    const srcDir = 'src';
    const output = execSync(
      `echo "$(cd ./${srcDir} && git grep -n -E "(@redhat-cloud-services/frontend-components/components/)")"`
    );

    expect(
      output
        .toString()
        .trim()
        .split(/[\n\r]/g)
        .filter(str => !/\/(cjs|esm)\//.test(str))
    ).toMatchSnapshot('direct import exceptions');
  });
});
