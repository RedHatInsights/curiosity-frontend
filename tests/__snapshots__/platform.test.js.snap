// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Platform Configuration, Build should have a name reference that matches configuration: ui name 1`] = `
Array [
  "REACT_APP_UI_NAME=subscriptions",
  "",
]
`;

exports[`Platform Configuration, Build should have a proxy configuration with path references that match configuration: proxy references 1`] = `
Array [
  "'/apps/subscriptions': {",
  "'/beta/apps/subscriptions': {",
  "'/staging/subscriptions': {",
  "'/beta/staging/subscriptions': {",
  "'/subscriptions': {",
  "'/beta/subscriptions': {",
  "'/insights/subscriptions': {",
  "'/beta/insights/subscriptions': {",
  "'/rhel/subscriptions': {",
  "'/beta/rhel/subscriptions': {",
  "'/openshift/subscriptions': {",
  "'/beta/openshift/subscriptions': {",
  "",
]
`;

exports[`Platform Configuration, Build should have path references that match configuration: path references 1`] = `
Array [
  "PUBLIC_URL=\${UI_DEPLOY_PATH_PREFIX}/apps/subscriptions/",
  "REACT_APP_CONFIG_SERVICE_LOCALES=\${UI_DEPLOY_PATH_PREFIX}/apps/subscriptions/locales/locales.json",
  "REACT_APP_CONFIG_SERVICE_LOCALES_PATH=\${UI_DEPLOY_PATH_PREFIX}/apps/subscriptions/locales/{{lng}}.json",
  "",
]
`;

exports[`Platform Configuration, Build should use direct imports for platform components, with exceptions: direct import exceptions, components 1`] = `Array []`;

exports[`Platform Configuration, Build should use direct imports for platform components, with exceptions: import exceptions, notifications 1`] = `Array []`;

exports[`Platform Configuration, Build should use direct imports for platform components, with exceptions: import exceptions, utilities 1`] = `Array []`;
